apps/events/models.py

```diff
@@ -0,0 +1,165 @@
+    )
+    status = models.CharField(choices=EventLogStatusChoices.choices)
+
+    class Meta:
+        indexes = [
+            models.Index(fields=["content_type", "object_id"]),
+        ]
+
+
+class StaticTriggerType(models.TextChoices):
+    CONVERSATION_END = "conversation_end"
+    LAST_TIMEOUT = "last_timeout"
+
+
+class StaticTrigger(BaseModel):
+    action = models.OneToOneField(EventAction, on_delete=models.CASCADE, related_name="static_trigger")
+    experiment = models.ForeignKey(Experiment, on_delete=models.CASCADE, related_name="static_triggers")
+    type = models.CharField(choices=StaticTriggerType.choices, db_index=True)
+    event_logs = GenericRelation(EventLog)
+
+    def fire(self, session):
+        try:
+            result = ACTION_FUNCTIONS[self.action.action_type](session, self.action.params)
+            self.event_logs.create(session=session, status=EventLogStatusChoices.SUCCESS)
+        except Exception:
+            self.event_logs.create(session=session, status=EventLogStatusChoices.FAILURE)
+
+        return result
+
+
+class TimeoutTrigger(BaseModel):
+    action = models.OneToOneField(EventAction, on_delete=models.CASCADE, related_name="timeout_trigger")
+    experiment = models.ForeignKey(Experiment, on_delete=models.CASCADE, related_name="timeout_triggers")
+    delay = models.PositiveIntegerField(
+        help_text="The amount of time in seconds to fire this trigger.",
+    )
+    total_num_triggers = models.IntegerField(
+        default=1,
+        help_text="The number of times to fire this trigger",
+    )
+    event_logs = GenericRelation(EventLog)
+
+    def timed_out_sessions(self):
+        """Finds all the timed out sessions where:
+        - The last human message was sent at a time earlier than the trigger time
+        - There have been fewer trigger attempts than the total number defined by the trigger
+        """
+
+        trigger_time = timezone.now() - timedelta(seconds=self.delay)
+
+        last_human_message_created_at = (
+            ChatMessage.objects.filter(
+                chat__experiment_session=OuterRef("pk"),
+                message_type=ChatMessageType.HUMAN,
+            )
+            .order_by("-created_at")
+            .values("created_at")[:1]
+        )
+        last_human_message_id = (
+            ChatMessage.objects.filter(
+                chat__experiment_session=OuterRef("session_id"),
+                message_type=ChatMessageType.HUMAN,
+            )
+            .order_by("-created_at")
+            .values("id")[:1]
+        )
+        log_count_for_last_message = (
+            EventLog.objects.filter(
+                session=OuterRef("pk"),
+                chat_message_id=Subquery(last_human_message_id),
+                status=EventLogStatusChoices.SUCCESS,
+            )
+            .annotate(
+                count=Func(F("chat_message_id"), function="Count")
```

Comment:
you should be able to use `django.db.models.Count`
```suggestion
                count=Count("chat_message_id")
```

Reply:
I added a comment about this on the next line. Unfortunately, we can't use `Count` because it breaks the subquery.

Reply:
Right, I thought we'd used it before but I see we didn't do a subquery, we did an "id__in" query